File: Practice2_Node/src/village_cheung/src/cheung_oop.cpp
#include "rclcpp/rclcpp.hpp"

class writernode :public rclcpp::Node
{
    public:
        writernode(std::string name):Node(name){

            RCLCPP_INFO(this->get_logger(),"大家好,我是%s",name.c_str());
        }
    private:
};

int main(int argc,char *argv[]){
    rclcpp::init(argc,argv);
    auto node = std::make_shared<writernode>("cheung");
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;

}

File: Practice2_Node/src/village_cheung_py/village_cheung_py/cheung_oop.py
import rclpy
from rclpy.node import Node

class writernode(Node):
    def __init__(self, name):
        super().__init__(name)
        self.get_logger().info("大家好,我是%s" %name) #格式化字串有很4種

def main(args=None):
    rclpy.init(args=args)
    node = writernode("cheung")
    rclpy.spin(node)
    rclpy.shutdown()







